src/java.base/share/classes/java/nio/channels/FileChannel.java

```diff
@@ -655,6 +659,11 @@ public abstract long transferTo(long position, long count,
      * operating systems can transfer bytes directly from the source channel
      * into the filesystem cache without actually copying them.  </p>
      *
+     * @apiNote
+     * This method only returns the number of bytes that were actually
+     * transferred, it will not return -1, e.g. in case of EOF of the
+     * {@code src} parameter.
+     *
```

Comment:
Looking at this again, I think it may be better if we just add a sentence to the end of the second paragraph of the method description. The second paragraph already covers the case that fewer bytes are transferred. I think it can be something simple like "No bytes are transferred if the source channel has reached end-of-stream". I suggest "end-of-stream" rather than "EOF" because that is the phrase used in the RBC javadoc.

Reply:
Hi Alan, Thanks for reviewing.
I will add the sentence you suggested at the end of second paragraph. 
Still think it's better to add an apiNote, although it's a little redundant, but it helps emphasize it, I saw several cases of misunderstanding of transferFrom. How do you think about it?

Reply:
Given the other verbiage I don't think that an `apiNote` is actually necessary.

Reply:
Hi Brian, 
At the beginning, I don't think this patch is necessary, because I think if I read really carefully, then I can avoid the misuse.
But the truth is there are several misuse cases, not just in my company, several are in opensource projects. So, I think it's worth to emphasize it even if it's a little redudant.
How do you think about it?

Reply:
The transferXXX methods return the number of bytes that were transferred to or from the file so there should be no expectation that -1 is returned. My preference would be to just clarify the transferFrom javadoc to make it clear that zero is returned when the source has reached end-of-stream.

Reply:
OK, I have removed the apiNote and only refine the javadoc as Brian suggested.
Would you mind to take another look? Thanks